api/src/error/errors.ts

```diff
@@ -0,0 +1,69 @@
+export enum AppErrorCode {
+  BAD_REQUEST,
+  UNAUTHENTICATED,
+  PERMISSIION_DENIED,
+  NOT_FOUND,
+  ALREADY_EXIST,
+  INTERNAL_ERROR
+}
+
+export enum HttpErrorCode {
+  BAD_REQUEST = 100,
+  UNAUTHORIZED,
+  FORBIDDEN,
+  NOT_FOUND,
+  METHOD_NOT_ALLOWED,
+  NOT_ACCEPTABLE,
+  PAYLOAD_TOO_LARGE,
+  UNSUPPORTED_MEDIA_TYPE,
+  INTERNAL_ERROR
+}
+
+export class CustomError extends Error {
+  messages: string[];
+
+  constructor(
+    public code: AppErrorCode | HttpErrorCode,
+    public origin: unknown,
+    ...messages: string[]
+  ) {
+    super(messages[0]);
+    Object.setPrototypeOf(this, CustomError.prototype);
+    this.messages = messages;
+  }
+
+  public codeToHttpStatusCode = () => {
```

Comment:
해당 메소드는 class 내부의 code 변수는 숨겨져있는 부분이라 메소드 이름에 `codeToXXX` 의 형태로 노출될 필요는 없을 것 같다는 생각이 드는데 어떻게 생각하실까요?

```suggestion
  public getHttpStatusCode = () => {
```

Reply:
말씀하신 내용이 맞는 것 같습니다. 클래스 내부 변수의 이름을 굳이 외부에 노출시킬 이유가 전혀 없네요ㅎㅎ

Reply:
https://github.com/MovieReviewComment/Mr.C/pull/73/commits/cde65ef0db3efffdde2b9ea6867424c02ba4f38d 에서 반영했습니다. 리뷰 감사합니다.

Reply:
수정 감사합니다~ 👍